packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -229,20 +265,8 @@ async function main() {
     }
 
     // VSCode extension prompt for installation
-    const alreadyPrompted = hasPromptedForHHVSCode();
-    if (!alreadyPrompted) {
-      const installationConsent = await confirmHHVSCodeInstallation();
-
-      if (installationConsent !== undefined) {
-        if (installationConsent) {
-          execSync("code --install-extension NomicFoundation.hardhat-solidity");
-        } else {
-          console.log(
-            "To learn more about Hardhat for Visual Studio Code, go to https://hardhat.org/hardhat-vscode"
-          );
-        }
-        writePromptedForHHVSCode();
-      }
+    if (taskName === "test") {
+      await suggestInstallingHardhatVscode();
```

Comment:
I think you are already aware of this, but just in case: this condition also needs to include `!isRunningOnCiServer() && process.stdout.isTTY === true`